addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -8394,10 +8394,17 @@ registry.many2one = SnippetOptionWidget.extend({
  * Allows to display a warning message on outdated snippets.
  */
 registry.VersionControl = SnippetOptionWidget.extend({
+    events: {
+        "click .o_snippet_replace": "_onReplaceSnippetClick",
+        "click .o_access_options": "_onAccessOptionsClick",
+    },
```

Comment:
Unless I am mistaken, we should not handle those buttons as custom ones but as standard UserValueWidget. Instead of rendering an alert with static XML by yourself, use `_renderCustomXML`. That way you will be able:
- Render the alert container
- Set up we-button elements inside, with standard data-XXX attributes which call XXX methods which handle async operations, history management, queued operations in the editor mutex, etc